soc/arm/silabs_exx32/Kconfig

```diff
@@ -81,6 +81,11 @@ config SOC_GECKO_RTCC
 	help
 	  Set if the Real Time Counter and Calendar (RTCC) HAL module is used.
 
+config SOC_GECKO_SE
+	bool
+	help
+	  Set if the Secure Element (SE) HAL module is used.
```

Comment:
@l-alfred is this related to TrustZone-M ?

Reply:
@ioannisg it is related to all security features available on the Gecko Series 2 SoC. It comprises a TRNG, hardware cryptographic acceleration elements (e.g., AES, SHA, ECC), the ARM TrustZone, a locking mechanism of the debug functionalities and some secure boot features. It works by sending commands to the SE mailbox (i.e., 64 word FIFO).

For example, It is used by the Silicon Labs mbedTLS implementation.

Reply:
Is the option used in Zephyr, though?

Reply:
Does this SoC implements the Security Extension, btw?

Reply:
> Is the option used in Zephyr, though?

I have an implementation of the entropy driver using the SE option. However, Silicon Labs recommends to use it through their mbedTLS implementation instead. That is why I did not push the driver, as I am not sure what is the best option. In any case, this will be needed for security-related topics why using the SoC, let it be the third-party mbedTLS implementation or a simple entropy driver.

Reply:
> Does this SoC implements the Security Extension, btw?

I am not sure I follow, what do you mean by Security Extension? Are you referring to the secure-privileged, non-secure-privileged, secure-unprivileged and non-secure-unprivileged states?

Reply:
I am asking if this SoC implements TrustZone-M (ARM Security Extension for the M profile). That's essential.

- If it does not, then, I don't see any issues.
- It if does, then there should be a selection of a Kconfig that indicates the SAU/IDAU presence. You should also be selecting ARM_TRUSTZONE_M. In that case, is it possible to run Zephyr as Secure firmware on this SoC?
- If TrustZone -M is present, but Zephyr is only supposed to run together with a secure FW counterpart, as a Non-Secure firmware, then, you should be enabling TRUSTED_EXECUTION_NONSECURE by default, and document this in the .rst.

Reply:
> I am asking if this SoC implements TrustZone-M (ARM Security Extension for the M profile). That's essential.
> 
>     * If it does not, then, I don't see any issues.
> 
>     * It if does, then there should be a selection of a Kconfig that indicates the SAU/IDAU presence. You should also be selecting ARM_TRUSTZONE_M. In that case, is it possible to run Zephyr as Secure firmware on this SoC?
> 
>     * If TrustZone -M is present, but Zephyr is only supposed to run together with a secure FW counterpart, as a Non-Secure firmware, then, you should be enabling TRUSTED_EXECUTION_NONSECURE by default, and document this in the .rst.

I see, although the SoC as the possibility to use the ARM TrustZone-M, this PR does not implement it.

Reply:
That's OK. Then I suppose Zephyr will run, by default in Secure mode and will not sue any TrustZone-M features. Thanks for clarifying.

Reply:
Indeed, by default it seems to run in Secure mode. I can try to have a look and implement it, but as I'm not an expert, it might take a while. I think it will come with another PR.